src/Hazelcast.Net.Tests/Clustering/MemberPartitionGroupServerTests.cs

```diff
@@ -105,24 +116,39 @@ await AssertEx.SucceedsEventually(()
             // Check if client1 is connected to the new member
             // cannot use the old member id since we can only either kill or create member
             Assert.That(client3.Cluster.Connections.Count, Is.EqualTo(1));
-            Assert.That(client3.Members.Select(p => p.Member.ConnectAddress), Contains.Item(nAddress3));
+            Assert.That(client3.Members.Where(p => p.IsConnected).Select(p => p.Member.ConnectAddress.ToString()), Contains.Item(address3));
 
             AssertClientOnlySees(client1, address1);
             AssertClientOnlySees(client2, address2);
             AssertClientOnlySees(client3, address3);
         }
 
+        private async Task AddMemberFor(string connectedAddress)
+        {
+            Member member;
+            var nAddress3 = NetworkAddress.Parse(connectedAddress);
+            while (true)
+            {
+                member = await AddMember();
+                var created = new NetworkAddress(member.Host, member.Port);
+                if (created == nAddress3) break;
+
+                await RemoveMember(member.Uuid);
+            }
+        }
+
         [TestCase(RoutingModes.MultiMember)]
         [TestCase(RoutingModes.SingleMember)]
         [TestCase(RoutingModes.AllMembers)]
         public async Task TestClientCollectsClusterEvents(RoutingModes routingMode)
         {
             var address1 = "127.0.0.1:5701";
             var address2 = "127.0.0.1:5702";
+            var addreses = new string[] { address1, address2 };
             var keyCount = 3 * 271;
 
             // create a client with the given routing strategy for catching the events.
-            var client1 = await CreateClient(RoutingStrategy.PartitionGroups, address1, routingMode);
+            var client1 = await CreateClient(RoutingStrategy.PartitionGroups, addreses, "client1", routingMode);
```

Comment:
Nit pick: typo in variable name `addreses`

Reply:
Fixed.